backend/app/usecase/authenticator/thirdpartyapp.go

```diff
@@ -0,0 +1,125 @@
+package authenticator
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/short-d/app/fw/crypto"
+	"github.com/short-d/app/fw/timer"
+	"github.com/short-d/short/backend/app/entity"
+	"github.com/short-d/short/backend/app/usecase/authenticator/payload"
+	"github.com/short-d/short/backend/app/usecase/authorizer"
+	"github.com/short-d/short/backend/app/usecase/keygen"
+	"github.com/short-d/short/backend/app/usecase/repository"
+)
+
+// ThirdPartyApp authenticates the identify of a third application.
+type ThirdPartyApp struct {
+	authorizer authorizer.Authorizer
+	tokenizer  crypto.Tokenizer
+	keyGen     keygen.KeyGenerator
+	timer      timer.Timer
+	apiKeyRepo repository.APIKey
+	appRepo    repository.App
+}
+
+// GetApp retrieves app information based on the credential provided.
+func (t ThirdPartyApp) GetApp(cred Credential) (entity.App, error) {
+	if cred.APIKey == nil {
+		return entity.App{}, nil
+	}
+	apiKeyStr := *cred.APIKey
+	tokenPayload, err := t.tokenizer.Decode(apiKeyStr)
+	if err != nil {
+		return entity.App{}, err
+	}
+
+	apiKeyPayload, err := payload.NewAPIKey(tokenPayload)
+	if err != nil {
+		return entity.App{}, errors.New("invalid api key")
```

Comment:
Just thinking if we should append the error message from `NewAPIKey` to give more context for the client.